lib/backend/etcdbk/etcd.go

```diff
@@ -315,6 +304,49 @@ func New(ctx context.Context, params backend.Params, opts ...Option) (*EtcdBacke
 	return b, nil
 }
 
+func (b *EtcdBackend) checkVersion(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Second*9)
+	results := make(chan error, len(b.cfg.Nodes))
+
+	var wg sync.WaitGroup
+	for _, nn := range b.cfg.Nodes {
+		wg.Add(1)
+		go func(n string) {
+			defer wg.Done()
+			status, err := b.clients.Next().Status(ctx, n)
+			if err != nil {
+				results <- trace.Wrap(err)
+				return
+			}
+
+			ver := semver.New(status.Version)
+			min := semver.New(teleport.MinimumEtcdVersion)
+			if ver.LessThan(*min) {
+				results <- trace.BadParameter("unsupported version of etcd %v for node %v, must be %v or greater",
+					status.Version, n, teleport.MinimumEtcdVersion)
+				return
+			}
+
+			results <- nil
+		}(nn)
+	}
+
+	// wait for results
+	var err error
+	for range b.cfg.Nodes {
+		err = <-results
```

Comment:
Should we select over both `results` and `ctx.Done()` here?

Reply:
this is blocking on an operation that is scoped to `ctx`, so I don't think it adds anything.